Library/ViewModels/RewardAddOnSelectionViewModel.swift

```diff
@@ -269,13 +269,89 @@ public final class RewardAddOnSelectionViewModel: RewardAddOnSelectionViewModelT
         )
       }
 
-    Signal.zip(project, baseReward, refTag)
+    let calculatedShippingTotal = Signal.combineLatest(
```

Comment:
I understand that there are similarities between this and the `PledgeViewModel` but instead of copy/pasting entire blocks of code you should look to create a method that can encapsulate some of this logic and plug it in to the relevant portions of both view models/

Reply:
I would also clarify w Laurie if the scope of this ticket is just for new pledges or for both new pledges and editing an existing pledge.

 If it's just new pledges then a lot of this logic becomes much simpler.